### PR TITLE
#7 - Allow the user to step through the battle turn-by-turn

### LINKED ISSUE #7: Allow the user to step through the battle turn-by-turn
Allow the _test-battle.js_ script to be given an argument to pause the game each turn.

The script would run normally (and play all the turns) when not given arguments.

When started like `node test-battle.js --wait`, have the script wait for the user each turn. The output might resemble this:

```
-----
Turn 10:
-----
MyHero tried to move South
MyHero owns 0 diamond mines
MyHero has 100 health
|---|---|---|---|---|
|   |   |   |   |   |
|---|---|---|---|---|
|   |   |   |   |   |
|---|---|---|---|---|
|   |D00|WWW|D01|H00|
|---|---|---|---|---|
|   |   |   |   |   |
|---|---|---|---|---|
|   |   |   |   |H01|
|---|---|---|---|---|
********

Press ENTER to continue...
```

It would be nice if when using the "wait" mode, the previous turn's map would be cleared away, showing only one turn at a time.

### PR DESCRIPTION
**Pull request by tenshi13**
- rejig test-battle.js logic to allow press enter to continue turn
- added helper function clearScreen (clears cli)

this is my attempt to resolve #7 
@Asuza - sorry... I have moved the code a fair bit... 
do tell if I've changed it too much, I was having trouble with the stdin

### CODE REVIEW

File: helpers.js [hunk 1]
```diff
@@ -246,4 +246,9 @@ helpers.findNearestTeamMember = function (gameData) {
     return pathInfoObject.direction;
 };
 
+// Clears the console's screen
+helpers.clearScreen = function() {
```
Comment (Asuza) [inline helpers.js]: I can see why you put `clearScreen` in this file, but would you please move it to _test-battle.js_? The _hero.js_ and _helper.js_ files should be completely modifiable by the player, so _test-battle.js_ shouldn't depend on them for much more than what move the hero is going to make.

File: test-battle.js [hunk 2]
```diff
@@ -35,56 +49,127 @@ var enemyMoveFunction = function (gameData, helpers) {
...
     if (hero.name === 'MyHero') {
-
       // Ask your hero brain which way it wants to move
         direction = heroMoveFunction(game, helpers);
     } else {
         direction = enemyMoveFunction(game, helpers);
     }
     console.log('-----');
-    console.log('Turn ' + i + ':');
+    console.log('Turn ' + turn + ':');
```
Comment (Asuza) [inline test-battle.js:107]: Nice change!

**Comment by Asuza** [comment]
It's very good overall! Please run it through

```
./node_modules/eslint/bin/eslint.js ./test-battle.js 
```

when you're done making that change. Thanks!

File: helpers.js (hunk 1 above)
Comment (tenshi13) [reply helpers.js]: Yeah man, sorry wasn't very sure where to put it, I'll move it back there then

File: test-battle.js (hunk 2 above)
Comment (tenshi13) [reply test-battle.js:107]: wooo very happy you liked it : D

**Pull request merged**